### PR TITLE
Adjusts renagade gun list and spawns the gun in a bag if one exists

### LINKED ISSUE #1473: Renegades start with weapon in hand
#### Brief description of the issue
Not really a bug or anything, I guess. It just seems kinda like a bit of an oversight... like, if you spawn in right next to someone and the first thing you have to do is stuff your gun into your bag and... yeah.
#### What you expected to happen
I don't know what I expected, it was my first time being a renegade!
#### What actually happened
started with gun in hand
#### Steps to reproduce
be renegade
#### Additional info:
 - **Server Revision**: Found using the "Show Server Revision" verb under the OOC tab.
 - **Anything else you may wish to add** (Location if it's a mapping issue, etc)

### PR DESCRIPTION
**Pull request by Yoshax**
-Adds more weapons
-Spawns the gun in a bag if the player has one

Fixes #1473

### CODE REVIEW

File: code/game/antagonist/station/renegade.dm [hunk 1]
```diff
@@ -49,9 +62,22 @@ var/datum/antagonist/renegade/renegades
...
+	var/list/slots = list (
+		"backpack" = slot_in_backpack,
+		"left hand" = slot_l_hand,
+		"right hand" = slot_r_hand,
+		)
+	for(var/slot in slots)
+		player.equip_to_slot(gun, slot)
+		if(gun.loc == player)
+			break
+	var/obj/item/weapon/storage/S = locate() in player.contents
```
Comment (Kelenius) [inline code/game/antagonist/station/renegade.dm]: This will happily put the gun into a medkit.

File: code/game/antagonist/station/renegade.dm [hunk 2]
```diff
@@ -49,9 +62,22 @@ var/datum/antagonist/renegade/renegades
...
+	if(S && istype(S))
+		gun.loc = S
```
Comment (PJB3005) [inline code/game/antagonist/station/renegade.dm]: Should be `forceMove()`

**Comment by Kelenius** [comment]
Okay, now that I see the whole thing: consider using weighted pick instead of list-in-list. Either
``pick(1; thingA, 5; thingB)``
or
``pickweight(thingA = 1, thingB = 5)``

File: code/game/antagonist/station/renegade.dm (hunk 1 above)
Comment (Yoshax) [inline code/game/antagonist/station/renegade.dm]: Not really a problem though, is it? Loadout already shoves stuff into medkits and suchlike, and it's better it goes into a medkit than into their hand on spawn.

File: code/game/antagonist/station/renegade.dm [hunk 3]
```diff
@@ -15,19 +15,32 @@ var/datum/antagonist/renegade/renegades
...
+		/obj/item/weapon/gun/energy/crossbow,
+		/obj/item/weapon/gun/energy/crossbow/largecrossbow,
+		/obj/item/weapon/gun/projectile/automatic,
+		/obj/item/weapon/gun/projectile/automatic/mini_uzi,
 		/obj/item/weapon/gun/projectile/automatic/c20r,
-		/obj/item/weapon/gun/projectile/deagle/camo,
+		/obj/item/weapon/gun/projectile/automatic/sts35,
+		/obj/item/weapon/gun/projectile/automatic/wt550,
+		/obj/item/weapon/gun/projectile/automatic/z8,
+		/obj/item/weapon/gun/projectile/automatic/as24,
```
Comment (Neerti) [inline code/game/antagonist/station/renegade.dm]: I'm fairly sure giving renegades (or anyone for that matter) an automatic shotgun with 24 slugs loaded and **burst fire** is a bad idea.

File: code/game/antagonist/station/renegade.dm [hunk 4]
```diff
@@ -49,9 +61,22 @@ var/datum/antagonist/renegade/renegades
...
+	if(islist(gun_type))
+		gun_type = pick(gun_type)
 
+	var/obj/item/gun = new gun_type(get_turf(player))
+	var/list/slots = list (
+		"backpack" = slot_in_backpack,
+		"left hand" = slot_l_hand,
+		"right hand" = slot_r_hand,
+		)
+	for(var/slot in slots)
```
Comment (Zuhayr) [inline code/game/antagonist/station/renegade.dm]: I believe there is already a proc that will iterate over all slots and attempt to equip it. One of the hotkeys uses the same behavior. If it isn't a proc in that case it should probably be made one.

**Comment by Yoshax** [comment]
To do:
- Change the pick to be nicer
- Change the equipping procedure

File: code/game/antagonist/station/renegade.dm (hunk 1 above)
Comment (Datraen) [inline code/game/antagonist/station/renegade.dm]: Working on a quick fix for this.

**Comment by Yoshax** [comment]
To do is done. Ready for review/merge.

**Comment by Datraen** [comment]
I wouldn't mind a list of lists, if we're going to do something along those lines, but there are items that I feel should be included in preexisting lists, such as the colt and normal revolvers with the revolver list, and the pump shotguns into the shotgun list.

**Comment by Yoshax** [comment]
I did it by damage and caliber. Anything put within a second list is essentially a copy of the other's in the list. Grouping them otherwise seems silly because you're vastly reducing the chance of getting one of those items.

**Comment by Datraen** [comment]
Let's move the beanbag version of the double barreled shotgun from the list and into the main choices, then? The beanbags are different damage-wise than the pellets that the sawn-off and other double barreled shotgun have.

**Comment by Yoshax** [comment]
Sure. Can do.

**Pull request merged**